### PR TITLE
CB-19964 Added cdp-prometheus and cdp-request-signer logs to watched dirs by fluentd

### PR DESCRIPTION
**Pull request by szabolcs-horvath**

### CODE REVIEW

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/cdp-prometheus.sls [hunk 1]
```diff
@@ -76,6 +76,13 @@ remove_prometheus_pwd:
   {%- endif %}
 {%- endif %}
 
+/var/log/cdp-prometheus/cdp-prometheus.log:
+  file.managed:
+    - user: "root"
+    - group: "root"
+    - makedirs: True
+    - mode: 644
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/resources/salt-common/salt/monitoring/cdp-prometheus.sls:84]: why is the necessary?

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2 [hunk 2]
```diff
@@ -9,7 +9,7 @@ After=network-online.target
 Type=simple
 User=root
 Group=root
-ExecStart=/opt/cdp-prometheus/prometheus \
+ExecStart=/bin/sh -c "/opt/cdp-prometheus/prometheus \
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2:12]: why is this change required?

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2 [hunk 3]
```diff
@@ -25,7 +25,8 @@ ExecStart=/opt/cdp-prometheus/prometheus \
      --storage.agent.no-lockfile \
      --config.file=/opt/cdp-prometheus/prometheus.yml \
      --web.config.file=/opt/cdp-prometheus/prometheus-web-config.yml \
-     --web.listen-address=:{{ monitoring.agentPort }}
+     --web.listen-address=:{{ monitoring.agentPort }} \
+     >>/var/log/cdp-prometheus/cdp-prometheus.log 2>>/var/log/cdp-prometheus/cdp-prometheus.log"
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2:29]: usually we do `2>&1` instead of redirrecting it to the same file as stdout

**Review by lacikaaa — CHANGES_REQUESTED** [review]

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/cdp-prometheus.sls (hunk 1 above)
Comment (szabolcs-horvath) [reply orchestrator-salt/src/main/resources/salt-common/salt/monitoring/cdp-prometheus.sls:84]: The start command would fail, if the log file didn't exist. (see the start command in the service's unit file cdp-prometheus.service.j2)

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2 (hunk 2 above)
Comment (szabolcs-horvath) [reply orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2:12]: The redirection of the stdout and stderr at the end of the command made the ExecStart command fail, and the service unable to start, so that's why I had to wrap it.

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2 (hunk 3 above)
Comment (szabolcs-horvath) [reply orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2]: fixed

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/cdp-prometheus.sls (hunk 1 above)
Comment (lacikaaa) [reply orchestrator-salt/src/main/resources/salt-common/salt/monitoring/cdp-prometheus.sls:84]: I don't see why would it fail. maybe the directory should be created but not the log file

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2 (hunk 3 above)
Comment (lacikaaa) [reply orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2]: how is the log file rotated/cleaned up?

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2 (hunk 2 above)
Comment (lacikaaa) [reply orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2:12]: please check systemd manual, according to this: https://stackoverflow.com/a/54232531 there is a more sophisticated way for redirecting output

Comment (szabolcs-horvath) [reply orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2:12]: The solution suggested there only works with v236+ versions of systemd. If I remember correctly only v217 is available on our vms.

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2 (hunk 3 above)
Comment (szabolcs-horvath) [reply orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2]: I added the logrotate config file for cdp-prometheus, and cdp-request-signer already has log rotation built into the service.

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2 (hunk 2 above)
Comment (szabolcs-horvath) [reply orchestrator-salt/src/main/resources/salt-common/salt/monitoring/systemd/cdp-prometheus.service.j2:12]: I double checked, and systemd v219 cannot use the StandardOutput field to log directly into a file. I indicated this in the commit description.

File: orchestrator-salt/src/main/resources/salt-common/salt/monitoring/cdp-prometheus.sls (hunk 1 above)
Comment (szabolcs-horvath) [reply orchestrator-salt/src/main/resources/salt-common/salt/monitoring/cdp-prometheus.sls]: I changed it so only the directory is created ahead of time.

**Review by lacikaaa — APPROVED** [review]

**Review by doktoric — APPROVED** [review]
